### ISSUE TITLE
requestPermission() make app crash on iOS

### ISSUE THREAD

**Issue by charlesRms** [chemerisuk/cordova-plugin-firebase-messaging#32]
when using requestPermission, I get the following error message : 

```
2018-08-02 19:26:00.171606+0200 [963:229605] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]'
*** First throw call stack:
(0x1c813b3d 0x1ba9b067 0x1c72b485 0x1c72b2ef 0x227e3d 0x228809 0x6272d 0x60f29 0x8a3467 0x8a3453 0x8a7d47 0x1c7cfd69 0x1c7cde19 0x1c7211af 0x1c720fd1 0x1decbb41 0x21aa9a53 0x29d63 0x1bf0e4eb)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

after that, app is totally frozen, and the user cannot do anything else.

I think that requestPermission is required on iOS, am I wrong ?
otherwise I get the following error when trying to get token : 

`5.4.1 - [Firebase/InstanceID][I-IID003009] Failed to fetch default token Error Domain=com.firebase.iid Code=1003 "(null)"`

**Comment by Shikate** [comment]
Hi Charles, i'm facing the same issue. 
Not sure if it's a configuration mistake or anything else.
Do you already solve the problem ?

**Comment by WoollyMittens** [comment]
I'm on cordova-plugin-firebase-messaging@1.2.1 and iOS crashes every time "requestPermission" is used with "attempt to insert nil object from objects".

**Comment by RTK** [comment]
I encountered the same issue, and found out, that the following code throws the exception:

```
NSData* deviceToken = [FIRMessaging messaging].APNSToken;

pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArrayBuffer:deviceToken];
```

I think when your firebase configuration does not retrieve an APNSToken (since you don't have any APNS certificate or Key set up in your firebase console), a positive plugin result with nil as argument for the messageArrayBuffer is provided, which ultimately leads to the error.

**Comment by RTK** [comment]
I created a fork https://github.com/RTK/cordova-plugin-firebase-messaging, which solves the issue by calling the error callback, when the apns token cannot retrieved. I will create a PR soon.

**Comment by chemerisuk** [comment]
@RTK thanks for the catch, I'll review your PR soon.

**Comment by chemerisuk** [comment]
Bugfix published in v1.3.2
